File: 0118-pascals-triangle/0118-pascals-triangle.java
class Solution {
    public static List<Integer> generateRow(int row){
        List<Integer> resRow = new ArrayList<>();
        resRow.add(1);
        long ans =1;
        for(int col=1; col<row; col++){
            ans = ans*(row-col);
            ans = ans/col;
            resRow.add((int)ans);
        }
        return resRow;
    }
    
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> res = new ArrayList<>();
        for(int i=1; i<=numRows; i++){
            res.add(generateRow(i));
        }
        return res;
    }
}

File: 0118-pascals-triangle/0118-pascals-triangle.py
def generateRows(i):
    res = [1]
    ans=1
    for j in range(1,i):
        ans*=(i-j)
        ans/=j
        res.append(int(ans))
    return res
    

class Solution:      
    
    def generate(self, numRows: int) -> List[List[int]]:
        res = []
        for i in range(1,numRows+1):
            res.append(generateRows(i))
        return res
        